src/Utils/request/handleResponse.ts

```diff
@@ -27,7 +27,7 @@ export default function handleResponse(
 
     // Handle session expiry
     if (error?.detail === "Authentication credentials were not provided.") {
-      notify?.Error({ msg: "Session expired. Please Login again." });
+      navigate(`/session-expired?redirect=${window.location.href}`);
```

Comment:
Add an additional check to see if we are not on the /session-expired page. Otherwise we might be stuck in a redirect loop 🤔